constellation-app/constellation
# Added Find All and zoom button to find and replace panel

### Prerequisites

- [X ] Reviewed the [checklist](CHECKLIST.md)

    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

A new 'Find and Zoom' button added to the Basic Find dialog to zoom to find the matching nodes and to zoom to the selected nodes as noted in #1912.
The action is handled by the zoom to selection plugin.  
![find_and_zoom](https://github.com/constellation-app/constellation/assets/63685177/a1c52fc8-11e2-45aa-bc9b-4d9462141c74)


### Alternate Designs

None considered.

### Why Should This Be In Core?

This find and zoom functionality seems to be a common workflow for users and this PR makes that workflow a little easier. 

### Benefits

Simplified user interaction with the find functionality.

### Possible Drawbacks

Additional button has been added and may cluster the GUI. 

### Verification Process

1. Open the Basic Find dialog in Constellation (Ctrl + f)
2. Enter a search string in the Find field.
3. Click 'Find All and Zoom'
4. Verify the graph view has zoomed into the nodes and edges that match the search string.

### Applicable Issues

#1912
